x-pack/plugins/synthetics/public/apps/synthetics/components/settings/global_params/add_param_flyout.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiButton,
+  EuiTitle,
+  EuiFlyoutFooter,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FormProvider } from 'react-hook-form';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { FETCH_STATUS, useFetcher } from '@kbn/observability-plugin/public';
+import { SavedObject } from '@kbn/core-saved-objects-common';
+import { i18n } from '@kbn/i18n';
+import { syntheticsParamType } from '../../../../../../common/types/saved_objects';
+import { useSyntheticsRefreshContext } from '../../../contexts';
+import { SyntheticsParam } from '../../../../../../common/runtime_types';
+import { useFormWrapped } from '../../../../../hooks/use_form_wrapped';
+import { AddParamForm } from './add_param_form';
+
+export const AddParamFlyout = ({
+  items,
+  isEditingItem,
+  setIsEditingItem,
+  setRefreshList,
+}: {
+  items: Array<SavedObject<SyntheticsParam>>;
+  setRefreshList: React.Dispatch<React.SetStateAction<number>>;
+  isEditingItem: SavedObject<SyntheticsParam> | null;
+  setIsEditingItem: React.Dispatch<React.SetStateAction<SavedObject<SyntheticsParam> | null>>;
+}) => {
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
+
+  const form = useFormWrapped<SyntheticsParam>({
+    mode: 'onSubmit',
+    reValidateMode: 'onChange',
+    shouldFocusError: true,
+    defaultValues: isEditingItem?.attributes ?? {
+      key: '',
+      tags: [],
+      description: '',
+      value: '',
+    },
+  });
+
+  const closeFlyout = useCallback(() => {
+    setIsFlyoutVisible(false);
+    setIsEditingItem(null);
+    form.reset({ key: '', tags: [], description: '', value: '' });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setIsEditingItem]);
+
+  const [paramData, setParamData] = useState<SyntheticsParam | null>(null);
+
+  const { savedObjects, application } = useKibana().services;
+
+  const { refreshApp } = useSyntheticsRefreshContext();
+
+  const { loading, status, data } = useFetcher(async () => {
+    if (paramData && savedObjects) {
+      if (isEditingItem) {
+        return savedObjects.client.update(syntheticsParamType, isEditingItem.id, paramData);
+      }
+      return await savedObjects.client.create(syntheticsParamType, paramData);
+    }
+  }, [paramData]);
+
+  const canSave = (application?.capabilities.uptime.save ?? false) as boolean;
+
+  const onSubmit = (formData: SyntheticsParam) => {
+    setParamData(formData);
+  };
+
+  useEffect(() => {
+    if (status === FETCH_STATUS.SUCCESS && data && !loading) {
+      closeFlyout();
+      setRefreshList(Date.now());
+      setParamData(null);
+    }
+  }, [data, loading, refreshApp, paramData, status, closeFlyout, setRefreshList]);
+
+  useEffect(() => {
+    if (isEditingItem) {
+      setIsFlyoutVisible(true);
+      form.reset(isEditingItem.attributes);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Can you please add a comment as to why you are disabling exhaustive deps. Disabling exhaustive deps often leads to difficult to debug bugs.